src/main/java/edu/kis/powp/jobs2d/drivers/DriverManager.java

```diff
@@ -1,35 +1,65 @@
 package edu.kis.powp.jobs2d.drivers;
 
+import edu.kis.legacy.drawer.panel.DrawPanelController;
+import edu.kis.legacy.drawer.shape.line.BasicLine;
 import edu.kis.powp.jobs2d.Job2dDriver;
-import edu.kis.powp.jobs2d.LoggerDriver;
+import edu.kis.powp.jobs2d.drivers.adapter.LineDriverAdapter;
+import edu.kis.powp.jobs2d.features.DriverFeature;
 import edu.kis.powp.observer.Publisher;
 
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
 /**
  * Driver manager provides means to setup the driver. It also enables other
  * components and features of the application to react on configuration changes.
  */
 public class DriverManager {
-
-    private Job2dDriver currentDriver = new LoggerDriver();
-
+    private LineDriverAdapter currentDriver = new LineDriverAdapter(new DrawPanelController(), new BasicLine(), "Setup driver");
+    private DriversComposite currentFeaturesComposite = new DriversComposite(this);
     private final Publisher changePublisher = new Publisher();
 
     /**
      * @param driver Set the driver as current.
      */
-    public synchronized void setCurrentDriver(Job2dDriver driver) {
+    public synchronized void setCurrentDriver(LineDriverAdapter driver) {
         currentDriver = driver;
         changePublisher.notifyObservers();
     }
 
+    public synchronized void updateDriversComposite(Job2dDriver driver) {
```

Comment:
names of methods are not precise. e.g. I think that _toggleFeature_ would be better in this case.